### ISSUE TITLE
Stalling under stress

### ISSUE THREAD

**Issue by markwylde** [Softmotions/ejdb#306]
Hi @adamansky 

Once again, loving ejdb. It's doing my project [canhazdb](https://github.com/canhazdb/server) so well. I've moved from sqlite to ejdb as the default driver. It's awesome.

I have come across one bug, I think. It took a few days to narrow it down to this simple example. I've tried as best I can looking into why this is happening, but I'm lost.

# Description of problem
When running multiple `q.list()` at the same time (and not many, as little as 5), on many occasions the promise will just not resolve.

# Versions, OS, etc...
| product             | version(s) |
| -------------------- |-------------|
| nodejs              | `v14.15.1`, `v12.20.1`, `v10.23.2` |
| ejdb2_node      | `2.0.581`      |
| linux                 | `ubuntu 20` |

# Recreate my problem
## Usage
```bash
mkdir ejdb2-stressed
cd ejdb2-stressed
npm init --yes
npm install ejdb2_node
# create index.js file from below
node index.js
```

The contents of `index.js` are:

```javascript
const fs = require('fs');
const { EJDB2 } = require('ejdb2_node');

const mapTimes = (times, fn) => Array(times).fill().map((_, index) => fn(index));

fs.rmdirSync('./data', { recursive: true });
fs.mkdirSync('./data');

let iterations = 0;
async function run (db) {
  iterations = iterations + 1;
  const iteration = iterations;

  // -----------------------------------------
  // PUT: Without any data it will not stall
  // -----------------------------------------
  console.time(`${iteration}:put`);
  await db.put('tests', { i: 1 });
  console.timeEnd(`${iteration}:put`);

  // -----------------------------------------
  // GET: This is the bit I think is stalling
  // -----------------------------------------
  console.time(`${iteration}:get`);
  const q = db.createQuery('/*', 'tests');

  await Promise.all(mapTimes(5, async () => {
    return q.list();
  }));
  console.timeEnd(`${iteration}:get`);

  // -----------------------------------------
  // REPEAT: Do it all again
  // -----------------------------------------
  run(db);
}

EJDB2.open('data/example.db')
  .then(db => run(db));
```

# What is the result?
When I run this, sometimes I get only 1 or 2 iterations, sometimes 15. But never more than 100, then it just hangs. Every time.

Here is an example of the output of my last run.

```text
node index.js 
1:put: 30.747ms
1:get: 8.888ms
2:put: 0.3ms
2:get: 1.771ms
3:put: 0.162ms
3:get: 1.527ms
4:put: 0.188ms
4:get: 1.547ms
5:put: 0.132ms
5:get: 1.268ms
6:put: 0.082ms
6:get: 1.374ms
7:put: 0.095ms
7:get: 1.465ms
8:put: 0.108ms
8:get: 1.168ms
9:put: 0.057ms
9:get: 1.311ms
10:put: 0.094ms
10:get: 1.36ms
11:put: 0.051ms
11:get: 1.27ms
12:put: 0.088ms
12:get: 1.274ms
13:put: 0.087ms
13:get: 1.371ms
14:put: 0.06ms
14:get: 1.375ms
15:put: 0.036ms
```

After the line `15:put: 0.036ms` the `q.list()` never resolves.

Would you have any ideas, or can point me in the right direction?

**Comment by adamansky** [comment]
Hi Mark! Thnak you for the report, it seems to be nodejs related bug. Will investigate it next week

**Comment by markwylde** [comment]
Hey @adamansky 

I've had a look through the typescript but it really looks like the binding simply stops streaming any more data.

Could you point me roughly in the right direction as to where to look?

Thanks again

**Comment by adamansky** [comment]
Hi Mark @markwylde I'm in travel right now,  will look into this problem in few days

**Comment by adamansky** [comment]
Confirmed

**Comment by markwylde** [comment]
Thanks for looking into this @adamansky Let me know if there is anything I can do to help.

**Comment by adamansky** [comment]
Thank you Mark for this detailed report, fixed in d766976. Tomorrow I will publish update to npm.

**Comment by markwylde** [comment]
Awesome, thanks again @adamansky

**Comment by adamansky** [comment]
Published 2.0.592 https://www.npmjs.com/package/ejdb2_node
